ethereum_history_api/circuits/lib/src/merkle_patricia_proofs/proof.nr

```diff
@@ -23,6 +25,45 @@ pub fn leaf_get_len<MAX_LEN>(leaf: Leaf<MAX_LEN>) -> u32 {
     leaf_unpadded_len as u32
 }
 
+// Prefixes mean:
+// 0 - Extension node, even number of nibbles
+// 1 - Extension node, odd number of nibbles
+// 2 - Leaf node, even number of nibbles
+// 3 - Leaf node, odd number of nibbles
+pub fn parity(prefix: u8) -> u8 {
+    assert(prefix < 4, "Prefix must be 0, 1, 2 or 3");
+    let result = prefix % 2;
+    assert((result == EVEN) | (result == ODD), "Invalid parity");
+
+    result
+}
+
+// Leaf and extension nodes are RLP lists with 2 elements: prefixed key nibbles (in bytes) and value.
+// This function removes the prefix and returns appropriate number of key nibbles according to the parity of the prefix (see function: parity).
+pub fn strip_prefix<MAX_KEY_LEN, MAX_KEY_NIBBLE_LEN>(
+    prefixed_key_bytes: [u8; MAX_KEY_LEN],
+    prefixed_len: u64
+) -> ([u8; MAX_KEY_NIBBLE_LEN], u64) {
+    assert(MAX_KEY_LEN * 2 == MAX_KEY_NIBBLE_LEN, "Generic arguments don't match");
+    let prefixed_key_nibbles: [u8; MAX_KEY_NIBBLE_LEN] = bytes_to_nibbles(prefixed_key_bytes);
+
+    let mut stripped_nibbles = [0; MAX_KEY_NIBBLE_LEN];
+    let parity = parity(prefixed_key_nibbles[0]);
+    if (parity == EVEN) {
+        assert(prefixed_key_nibbles[1] == 0, "Expected a zero after a prefix of even parity");
+    }
+    let offset = 2 - parity as u64;
```

Comment:
Let's make returned value from `parity` the same type as offset, so we don't need to convert it. 
`parity` function is only used here